### PR TITLE
Asynchronous task launch

### PR DESCRIPTION
**Pull request by ewjoachim**
Cf. #9 
Co-Authored by @sophie-ulti and @pmourlanne 

### Successful PR Checklist:
- [ ] Tests
- [ ] Documentation
- [ ] Had a good time contributing? (if not, feel free to give some feedback)

### CODE REVIEW

File: procrastinate/jobs.py [hunk 1]
```diff
@@ -65,21 +67,57 @@ class JobLauncher:
...
     def defer(self, **task_kwargs: types.JSONValue) -> int:
         """
         See :py:func:`Task.defer` for details.
         """
-        final_kwargs = self.job.task_kwargs.copy()
-        final_kwargs.update(task_kwargs)
+        from procrastinate import store as store_module
 
-        job = attr.evolve(self.job, task_kwargs=final_kwargs)
+        assert isinstance(
```
Comment (ewjoachim) [inline procrastinate/jobs.py]: job_store.asynchronous

Comment (ewjoachim) [reply procrastinate/jobs.py]: No, but add a comment saying it's for mypy

**Comment by ewjoachim** [comment]
Still to write doc.

**Comment by ewjoachim** [comment]
@pmourlanne Ready for final review.

Ping @mgu too.

File: tests/acceptance/test_async.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import pytest
+
+import procrastinate
+
+pytestmark = pytest.mark.asyncio
```
Comment (pmourlanne) [inline tests/acceptance/test_async.py]: We (I :>) should look into doing this better :O There should be a fixture that lets us mark a test as `asyncio`. Then "asynchronous fixtures" like `async_app` could use this `asyncio` fixture. Thus we would not have to explicitly mark asynchronous tests.
The `db` fixture in `pytest_django` works like that. It's an empty shell that just adds the marker to the test.
(There's no call to action here, just the ramblings of a future old fool :santa: )

Comment (pmourlanne) [reply tests/acceptance/test_async.py]: Linked issue: https://github.com/peopledoc/procrastinate/issues/95

File: procrastinate/testing.py [hunk 3]
```diff
@@ -10,24 +10,17 @@
...
         """
         Attributes
         ----------
-        jobs : List[jobs.Job]
-            Currently running Job objects. They are removed from this
-            list upon completion
-        finished_jobs: List[Tuple[jobs.Job, jobs.Status]]
-            List of finished jobs, with their related status
+        jobs : Dict[int, Dict]
+            Mapping of ``{<job id>: <Job database row as a dictionary>}``
```
[the remark this answers is not in the document]
Comment (ewjoachim) [reply procrastinate/testing.py]: finished_jobs doesn't exist anymore. This should have been done long ago.

**Comment by ewjoachim** [comment]
@pmourlanne comments integrated. All good.

**Pull request merged**